src/S2EExecutionState.cpp

```diff
@@ -336,7 +336,7 @@ std::vector<ref<Expr>> S2EExecutionState::createConcolicArray(const std::string
     // Add it to the set of symbolic expressions, to be able to generate
     // test cases later.
     // Dummy memory object
-    MemoryObject *mo = new MemoryObject(0, size, false, false, false, NULL);
+    MemoryObject *mo = new MemoryObject(address, size, false, false, false, NULL);
```

Comment:
IIRC, KLEE assumes that this is the host address, while you pass a guest address. It might create problems.
It might be better to add a guest_address field to MemoryObject, or add some sort of enum to caracterize the type of address.